src/audio/SDL_audiocvt.c

```diff
@@ -157,15 +157,96 @@ static SDL_bool SDL_IsSupportedChannelCount(const int channels)
     return ((channels >= 1) && (channels <= 8));
 }
 
+SDL_bool SDL_ChannelMapIsBogus(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = 0; i < channels; i++) {
+            if (map[i] >= ((Uint8) channels)) {
+                return SDL_TRUE;
+            }
+        }
+    }
+    return SDL_FALSE;
+}
+
+SDL_bool SDL_ChannelMapIsDefault(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = 0; i < channels; i++) {
+            if (map[i] != i) {
+                return SDL_FALSE;
+            }
+        }
+    }
+    return SDL_TRUE;
+}
+
+SDL_bool SDL_ChannelMapHasDuplicates(const Uint8 *map, int channels)
+{
+    if (map) {
+        for (int i = channels - 1; i >= 1; i--) {
+            for (int j = i - 1; j >= 0; j--) {
+                if (map[i] == map[j]) {
+                    return SDL_TRUE;
+                }
+            }
+        }
+    }
+
+    return SDL_FALSE;
+}
+
+// Swizzle audio channels. src and dst can be the same pointer. It does not change the buffer size.
+static void SwizzleAudio(const int num_frames, void *dst, const void *src, int channels, const Uint8 *map, int bitsize)
+{
+    const SDL_bool allow_inplace = (src != dst) || (!SDL_ChannelMapHasDuplicates(map, channels));
```

Comment:
I don't think the absence of duplicates means you can safely do this inplace when src==dst.
I.E consider trying to swap the L/R channels without a temporary variable.

Reply:
Oh, yeah, good point. I've taken this out in 136a33153cc797eeed2b46600638c76bb064e6a1.